pinc/Project.inc

```diff
@@ -1030,14 +1030,11 @@ class Project
 
     public function get_num_pages_in_state($state = null, $counter_sql = "*")
     {
-        if (!$this->pages_table_exists) {
-            throw new NoProjectPageTable(_("Project page table does not exist."));
-        }
+        $this->validate_project_page_table();
 
         // if state is null, get all pages
         $where = $state ? sprintf("state = '%s'", DPDatabase::escape($state)) : "1";
 
-        validate_projectID($this->projectid);
```

Comment:
We should continue to do this before we use it in the SQL rather than moving it in a new function (we can also have it in the new function if you want). Having it before the SQL where we use is a reinforcement that we are explicitly validating variables in SQL before using them. We've had unvalidated `$projectid` be used in a security incident before so this is something I'm very aware of.

Reply:
ok